src/main/java/appeng/core/sync/packets/PacketNEIRecipe.java

```diff
@@ -62,117 +61,297 @@
+	//
+	// if( inv != null && this.recipeStack != null && security != null )
+	// {
+	// InventoryCrafting testInv = new InventoryCrafting( new ContainerNull(), 3, 3 );
+	// for( int x = 0; x < 9; x++ )
+	// {
+	// if( this.recipeStack[x] != null && this.recipeStack[x].length > 0 )
+	// {
+	// testInv.setInventorySlotContents( x, this.recipeStack[x][0] );
+	// }
+	// }
+	//
+	// IRecipe r = Platform.findMatchingRecipe( testInv, pmp.worldObj );
+	//
+	// if( r != null && security.hasPermission( player, SecurityPermissions.EXTRACT ) )
+	// {
+	// ItemStack is = r.getCraftingResult( testInv );
+	//
+	// if( is != null )
+	// {
+	// IMEMonitor<IAEItemStack> storage = inv.getItemInventory();
+	// IItemList all = storage.getStorageList();
+	// IPartitionList<IAEItemStack> filter = ItemViewCell.createFilter( cct.getViewCells() );
+	//
+	// for( int x = 0; x < craftMatrix.getSizeInventory(); x++ )
+	// {
+	// ItemStack patternItem = testInv.getStackInSlot( x );
+	//
+	// ItemStack currentItem = craftMatrix.getStackInSlot( x );
+	// if( currentItem != null )
+	// {
+	// testInv.setInventorySlotContents( x, currentItem );
+	// ItemStack newItemStack = r.matches( testInv, pmp.worldObj ) ? r.getCraftingResult( testInv ) : null;
+	// testInv.setInventorySlotContents( x, patternItem );
+	//
+	// if( newItemStack == null || !Platform.isSameItemPrecise( newItemStack, is ) )
+	// {
+	// IAEItemStack in = AEItemStack.create( currentItem );
+	// if( in != null )
+	// {
+	// IAEItemStack out = realForFake == Actionable.SIMULATE ? null : Platform.poweredInsert( energy, storage, in,
+	// cct.getSource() );
+	// if( out != null )
+	// {
+	// craftMatrix.setInventorySlotContents( x, out.getItemStack() );
+	// }
+	// else
+	// {
+	// craftMatrix.setInventorySlotContents( x, null );
+	// }
+	//
+	// currentItem = craftMatrix.getStackInSlot( x );
+	// }
+	// }
+	// }
+	//
+	// // True if we need to fetch an item for the recipe
+	// if( patternItem != null && currentItem == null )
+	// {
+	// // Grab from network by recipe
+	// ItemStack whichItem = Platform.extractItemsByRecipe( energy, cct.getSource(), storage, player.worldObj, r, is,
+	// testInv, patternItem, x, all, realForFake, filter );
+	//
+	// // If that doesn't get it, grab exact items from network (?)
+	// // TODO see if this code is necessary
+	// if( whichItem == null )
+	// {
+	// for( int y = 0; y < this.recipeStack[x].length; y++ )
+	// {
+	// IAEItemStack request = AEItemStack.create( this.recipeStack[x][y] );
+	// if( request != null )
+	// {
+	// if( filter == null || filter.isListed( request ) )
+	// {
+	// request.setStackSize( 1 );
+	// IAEItemStack out = Platform.poweredExtraction( energy, storage, request, cct.getSource() );
+	// if( out != null )
+	// {
+	// whichItem = out.getItemStack();
+	// break;
+	// }
+	// }
+	// }
+	// }
+	// }
+	//
+	// // If that doesn't work, grab from the player's inventory
+	// if( whichItem == null && playerInventory != null )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)